src/Core/Checkout/Promotion/Aggregate/PromotionDiscount/PromotionDiscountEntity.php

```diff
@@ -228,19 +228,25 @@ public function getSetGroupId(): string
         return str_replace($prefix, '', $this->scope);
     }
 
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will return `?string` in the future
+     */
     public function getSorterKey(): string
     {
-        return $this->sorterKey;
+        return $this->sorterKey ?? '';
     }
 
     public function setSorterKey(string $sorterKey): void
     {
         $this->sorterKey = $sorterKey;
     }
 
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will return `?string` in the future
+     */
     public function getApplierKey(): string
     {
-        return $this->applierKey;
+        return $this->applierKey ?? '';
```

Comment:
s.a. - Is there a test that covers what happens when an empty string is used in a calculation?